spec/outputs/datadog_metrics_spec.rb

```diff
@@ -1 +1,83 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/datadog_metrics"
+
+describe LogStash::Outputs::DatadogMetrics do
+  let(:http_instance) { double('net_http_instance') }
+  let(:request) { double('net_http_request') }
+  let(:response) { double('net_http_response') }
+
+  let(:json_obj) { double('json_obj') }
+  let(:logger) { double('logger') }
+  let(:ddtags) { ['foo', 'bar'] }
+  let(:event_hash) do
+    {
+      'message' => 'Hello world!',
+      'source' => 'datadog-metrics',
+      'type' => 'generator',
+      'host' => 'localhost',
+      '@timestamp' => LogStash::Timestamp.now
+    }
+  end
+  let(:event)  { LogStash::Event.new(event_hash) }
+  let(:params) do
+    {
+      'api_key' => '123456789',
+      'dd_tags' => ddtags,
+      'metric_type' => 'gauge'
+    }
+  end
+
+  before do
+    allow(Net::HTTP).to receive(:new).and_return(http_instance)
+    allow(Net::HTTP::Post).to receive(:new).and_return(request)
+  end
+
+  subject { described_class.new(params) }
+
+  before do
+    expect(http_instance).to receive(:use_ssl=)
+    expect(http_instance).to receive(:verify_mode=)
+    subject.register
+  end
+
+  context 'construction' do
+    it 'register method calls collaborator methods' do
```

Comment:
For the plugins, IMO, construction == initialize + register
The expectations for the Net::HTTP instance are set in the before block - method `register` builds an ivar of Net::HTTP instance, therefore Net::HTTP is an external collaborator.

IMO I would make the Stud::Buffer a discrete object and not a mixin because 1) its internals are separately tested and 2) it would become a collaborator and one can assert that it was send messages as part of `register` and `receive`